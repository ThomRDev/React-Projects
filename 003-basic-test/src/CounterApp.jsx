import PropTypes from "prop-types"

import { useState } from "react"

const CounterApp = ({ initValue }) => {

    const [counter, setCounter] = useState(initValue)

    console.log("renderizo");
    const handleAdd = (event) => {
        // setCounter(counter + 1)
        setCounter(oldValue => oldValue + 1)
    }
    const handleSubtract = (event) => {
        if (counter <= 0) return
        // setCounter(counter - 1)
        setCounter(oldValue => oldValue - 1)
    }

    const handleReset = (event) => {
        setCounter(initValue)
    }
    return (
        <div className="App">
            <header className="App-header">
                <div className="value">{counter}</div>
                <div className='actions'>
                    <button onClick={handleAdd}>+1</button>
                    <button onClick={handleReset}>Reset</button>
                    <button onClick={handleSubtract}>-1</button>
                </div>
            </header>
        </div>
    );
}

CounterApp.propTypes = {
    initValue: PropTypes.number
}
CounterApp.defaultProps = {
    initValue: 1
}

export default CounterApp;