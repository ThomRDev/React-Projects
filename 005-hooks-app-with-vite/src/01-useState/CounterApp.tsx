import { useState } from "react"

export const CounterApp = () => {
  const [counter, setCounter] = useState({
    counter1:10,
    counter2:20,
    counter3:30,
  })
  return (
    <>
      <h2>Counter : { JSON.stringify(counter) }</h2>
      <hr />
      <button className="btn btn-primary" onClick={()=>setCounter({
        ...counter,
        counter1 : counter.counter1 + 1
      })} >+1 Counter1</button>
    </>
  )
}
