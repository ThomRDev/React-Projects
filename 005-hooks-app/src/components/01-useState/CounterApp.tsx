import React, { useState } from 'react'

const CounterApp = () => {
    const [{ counter1,counter2 }, setCounter] = useState({
        counter1 : 0,
        counter2 : 0
    })

    const handleActionClickCounter1 = (event:React.MouseEvent<HTMLButtonElement>) =>{
        event.preventDefault()
        const {action} = event.currentTarget.dataset
        if(action === "decrement"){
            setCounter(oldValue=>({
                ...oldValue,
                counter1 : counter1 - 1 
            }))
        }
    }
    const handleActionClickCounter2 = (event:React.MouseEvent<HTMLButtonElement>) =>{
        event.preventDefault()
        const {action} = event.currentTarget.dataset
        if(action === "decrement"){
            setCounter(oldValue=>({
                ...oldValue,
                counter2 : counter2 - 1 
            }))
        }
    }
    return (
        <div className='d-flex align-items-center gap-5'>  
            <section>
                <h2>Counter-1 : { counter1 }</h2>
                <div className="d-flex align-items-center gap-2">
                    <button onClick={ handleActionClickCounter1 } className="btn btn-primary d-grid fs-5" style={{placeItems:"center"}} data-action="increment" data-item="counter1">+1</button>
                    <button onClick={ handleActionClickCounter1 } className="btn btn-danger d-grid fs-5" style={{placeItems:"center"}} data-action="decrement" data-item="counter1">-1</button>
                    <button onClick={ handleActionClickCounter1 } className="btn btn-secondary d-grid fs-5" style={{placeItems:"center"}} data-action="reset" data-item="counter1">Reset</button>
                </div>
            </section>
            <section>
                <h2>Counter-2 : { counter2 }</h2>
                <div className="d-flex align-items-center gap-2">
                    <button onClick={ handleActionClickCounter2 } className="btn btn-primary d-grid fs-5" style={{placeItems:"center"}} data-action="increment" data-item="counter2">+1</button>
                    <button onClick={ handleActionClickCounter2 } className="btn btn-danger d-grid fs-5" style={{placeItems:"center"}} data-action="decrement" data-item="counter2">-1</button>
                    <button onClick={ handleActionClickCounter2 } className="btn btn-secondary d-grid fs-5" style={{placeItems:"center"}} data-action="reset" data-item="counter2">Reset</button>
                </div>
            </section>
            
        </div>
    )
}

export default CounterApp