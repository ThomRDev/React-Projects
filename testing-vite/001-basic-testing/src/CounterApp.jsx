import PropTypes from "prop-types"

import { useState } from "react"

export const CounterApp = ({ initValue }) => {

    const [counter, setCounter] = useState(initValue)

    const handleAdd = (event) => {
        setCounter(oldValue => oldValue + 1)
    }
    const handleSubtract = (event) => {
        if (counter <= 0) return
        setCounter(oldValue => oldValue - 1)
    }

    const handleReset = (event) => {
        setCounter(initValue)
    }
    return (
        <div className="App">
            <header className="App-header">
                <div className="value">{counter}</div>
                <div className='actions'>
                    <button onClick={handleAdd}>+1</button>
                    <button onClick={handleReset}>Reset</button>
                    <button onClick={handleSubtract}>-1</button>
                </div>
            </header>
        </div>
    );
}

CounterApp.propTypes = {
    initValue: PropTypes.number
}
CounterApp.defaultProps = {
    initValue: 1
}
