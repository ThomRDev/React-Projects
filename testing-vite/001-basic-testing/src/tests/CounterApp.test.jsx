import { render,screen,fireEvent } from "@testing-library/react"
import { CounterApp } from "../CounterApp"
describe('Test in CounterApp component', () => {
  test('should do to match with the snapshot', () => {
    const { container } = render( <CounterApp /> )
    expect(container).toMatchSnapshot()
  })
  test('should show the initial value = 100', () => {
    const {container,getByText} = render( <CounterApp initValue={100} /> )
    expect(getByText(100)).toBeTruthy()
    expect(container.querySelector(".value").innerHTML).toBe(100+"")
  })
  test('should increment with the button +1', () => {
    render( <CounterApp initValue={100} /> )
    fireEvent.click(screen.getByText("+1"))
    
    expect(screen.getByText("101")).toBeTruthy()
  })
  test('should subtract with the button -1', () => {
    render( <CounterApp initValue={100} /> )
    fireEvent.click(screen.getByText("-1"))
    
    expect(screen.getByText("99")).toBeTruthy()
  })

  test('should work the reset button', () => {
    render( <CounterApp initValue={100} /> )
    fireEvent.click(screen.getByText("-1"))
    fireEvent.click(screen.getByText("+1"))
    fireEvent.click(screen.getByText("+1"))
    fireEvent.click(screen.getByText("+1"))
    fireEvent.click(screen.getByText("-1"))

    fireEvent.click(screen.getByText("Reset"))
    
    expect(screen.getByText("100")).toBeTruthy()
  })
  
})

// mucha ayuda
// https://testing-playground.com/