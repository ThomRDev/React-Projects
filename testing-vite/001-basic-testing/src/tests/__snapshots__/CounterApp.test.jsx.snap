// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in CounterApp component should do to match with the snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        class="value"
      >
        1
      </div>
      <div
        class="actions"
      >
        <button>
          +1
        </button>
        <button>
          Reset
        </button>
        <button>
          -1
        </button>
      </div>
    </header>
  </div>
</div>
`;
